libr/util/strbuf.c

```diff
@@ -119,6 +120,17 @@ R_API bool r_strbuf_slice(RStrBuf *sb, int from, int len) {
 	return true;
 }
 
+R_API bool r_strbuf_setptr(RStrBuf *sb, char *s, int len) {
+	r_return_val_if_fail (sb, false);
+	if (len < 0) {
+		len = strlen (s);
+	}
+	sb->ptr = s;
+	sb->ptrlen = len;
```

Comment:
Something like this maybe?
```
sb->ptrlen = len < 0 ? strlen (s) + 1 : len;
```
Though then we have to be careful not to do something like `r_strbuf_setptr(sb, "fuck", 4)` but to always pass -1 if it's supposed to be a string vs. bytes.

Reply:
yeah sgtm

Reply:
but y our example is wrong, because the string is 5 char length . you missed the \0, thats why the strlen() is there to avoid this common human problem

Reply:
This is exactly what I am saying. Most functions that take args like `char *str, int len` where you can set len to -1 expect len to be either -1 or `strlen(str)` which would be 4 here. But here `r_strbuf_setptr(sb, "fuck", 4)` is wrong so you have to pass -1 (or 5) and this has to be explicitly specified/documented.

Reply:
well, the idea is that string functions may get no length at all, and bin ones must specify that

Reply:
That's what I am saying.